js/tab.js

```diff
@@ -171,6 +180,32 @@ define([
     return syntaxValue;
   }
   
+  Tab.prototype.getWorker = function() {
+    return new Promise((resolve, reject) => {
+      // if worker is off, return null
+      if (!Settings.get("user").useWorker) return resolve(null);
+      
+      var checkLimit = 20;
+      var findWorker = () => {
+        // if worker was turned while we were waiting on it, return null
+        if (!Settings.get("user").useWorker) return resolve(null);
+      
+        // return the worker if we found it
+        if (this.$worker) {
+          return resolve(this.$worker);
+        }
+        
+        // see if we've checked too many times
+        if (checkLimit-- < 0) return reject(new Error('Could not find worker.'));
+        
+        // otherwise, go to sleep and then check again
+        setTimeout(findWorker, 100);
```

Comment:
I'll look more into it tomorrow, or when a get a few minutes, that setTimeout does feel pretty hackey. I am probably missing something, but it seemed to me that `session/tab.$worker` was `null` when it aught to have been an instance of `Worker`.